src/centralized_telescope/init.rs

```diff
@@ -0,0 +1,114 @@
+//! ALBA's bounded DFS scheme setup functions
+
+use super::params::Params;
+use super::setup::Setup;
+
+use std::f64::consts::LOG2_E;
+
+/// Setup algorithm taking a Params as input and returning setup parameters (u,d,q)
+pub fn make_setup(params: &Params) -> Setup {
+    let set_size_f64 = params.set_size as f64;
+    let lower_bound_f64 = params.lower_bound as f64;
+
+    let proof_size_f64 = {
+        let numerator =
+            params.soundness_param + params.completeness_param.log2() + 5.0 - LOG2_E.log2();
+        let denominator = (set_size_f64 / lower_bound_f64).log2();
+        (numerator / denominator).ceil()
+    };
+
+    let ratio = 9.0 * set_size_f64 * LOG2_E / ((17.0 * proof_size_f64).powi(2));
+    let s1 = ratio - 7.0;
+    let s2 = ratio - 2.0;
+
+    if s1 < 1.0 || s2 < 1.0 {
+        // Small case, i.e. set_size <= λ^2
+        param_small_case(params, proof_size_f64)
+    } else {
+        let completeness_param2 = params.completeness_param.min(s2);
+        if proof_size_f64 < completeness_param2 {
+            // Case 3, Theorem 14, i.e.  set_size >= λ^3
+            param_high_case(params, proof_size_f64, completeness_param2)
+        } else {
+            // Case 2, Theorem 13, i.e. λ^3 > set_size > λ^2
+            param_mid_case(params, proof_size_f64, s1)
+        }
+    }
+}
+
+fn param_small_case(params: &Params, proof_size_f64: f64) -> Setup {
```

Comment:
I'd provide a comment to explain what `small`, `mid` and `high` proof means? If possible, we can even try to give more descriptive names but at least an explanation in the code is important.

Reply:
There should be comments in the code above in `make_setup` (c.f. written below). Do you think they are not enough?
```
    if s1 < 1.0 || s2 < 1.0 {
        // Small case, i.e. set_size <= λ^2
        param_small_case(params, proof_size_f64)
    } else {
        let completeness_param2 = params.completeness_param.min(s2);
        if proof_size_f64 < completeness_param2 {
            // Case 3, Theorem 14, i.e.  set_size >= λ^3
            param_high_case(params, proof_size_f64, completeness_param2)
        } else {
            // Case 2, Theorem 13, i.e. λ^3 > set_size > λ^2
            param_mid_case(params, proof_size_f64, s1)
        }
    }
```

Reply:
We would give descriptive documentation about the parameter cases. @rrtoledo let's open an issue for that

Reply:
@curiecrypt  If it is just adding a few comments, we can do it here no? I have already added quite a few in `prove_index` and so on afterall.